modules/tinymce/src/core/test/ts/browser/keyboard/PageUpDownKeysTest.ts

```diff
@@ -0,0 +1,89 @@
+import { Keys } from '@ephox/agar';
+import { context, describe, it } from '@ephox/bedrock-client';
+import { TinyAssertions, TinyContentActions, TinyHooks, TinySelections } from '@ephox/mcagar';
+import { PlatformDetection } from '@ephox/sand';
+
+import Editor from 'tinymce/core/api/Editor';
+import Theme from 'tinymce/themes/silver/Theme';
+
+describe('browser.tinymce.core.keyboard.PageUpDownKeyTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    add_unload_trigger: false,
+    base_url: '/project/tinymce/js/tinymce',
+    indent: false
+  }, [ Theme ], true);
+  const platform = PlatformDetection.detect();
+
+  context('Page Up', () => {
+    it('caret is at the beginning of an inline element, escape outside the node', () => {
```

Comment:
Since this will depend on browser default behavior for the non inline boundary case it would be hard to automate without say selenium. So instead you could place the caret at `<a href="#">|text</a>` and verify that it gets moved to `|<a href="#">text</a>` on key up since that is what the code will be doing in that case. And you could keypress in the middle of the link and assert that the caret was not moved at all to make sure that the native handling is not overriden.

Reply:
It's no possible to simulate the behavior only with key up because the changes implemented above requires key down - key up